include/irr/asset/IImageAssetHandlerBase.h

```diff
@@ -31,10 +37,207 @@ class IImageAssetHandlerBase : public virtual core::IReferenceCounted
+			auto referenceImageParams = referenceImage->getCreationParameters();
+			auto referenceRegions = referenceImage->getRegions();
+			auto referenceTopRegion = referenceRegions.begin();
+
+			core::smart_refctd_ptr<ICPUImage> newImage;
+			{
+				auto newImageParams = referenceImageParams;
+				newImageParams.arrayLayers = 1;
+				newImageParams.mipLevels = 1;
+				newImageParams.type = IImage::ET_2D;
+
+				auto newRegions = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<ICPUImage::SBufferCopy>>(1);
+				newRegions->front() = *referenceTopRegion;
+
+				const auto texelOrBlockByteSize = asset::getTexelOrBlockBytesize(referenceImageParams.format);
+				auto texelBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(texelOrBlockByteSize * newRegions->front().bufferRowLength * newImageParams.extent.height * newImageParams.extent.depth);
+
+				newImage = ICPUImage::create(std::move(newImageParams));
+				newImage->setBufferAndRegions(std::move(texelBuffer), newRegions);
+			}
+
+			using COPY_FILTER = CCopyImageFilter;
+			COPY_FILTER copyFilter;
+			COPY_FILTER::state_type state;
+
+			auto newTopRegion = newImage->getRegions().begin();
+
+			state.inImage = referenceImage.get();
+			state.outImage = newImage.get();
+			state.inOffset = { 0, 0, 0 };
+			state.outOffset = { 0, 0, 0 };
+			state.inBaseLayer = 0;
+			state.outBaseLayer = 0;
+			state.extent = newTopRegion->getExtent();
+			state.layerCount = 1;
+			state.inMipLevel = newTopRegion->imageSubresource.mipLevel;
+			state.outMipLevel = 0;
+
+			if(!copyFilter.execute(&state))
+				os::Printer::log("Something went wrong while copying top level region texel's data to the image!", ELL_WARNING);
+
+			return newImage;
+		}
+
+		/*
+			Create an image containing a single row from taking a raw data 
+			to a single row and convert it to any format. Since it's
+			data may not only limit to stuff being displayed on a screen,
+			there is an optiomal parameter for bufferRowLength pitch that
+			is helpful while dealing with specific data which needs it.
+		*/
+
+		template<E_FORMAT inputFormat, E_FORMAT outputFormat>
+		static inline core::smart_refctd_ptr<ICPUImage> createSingleRowImageFromRawData(void* rowData, uint32_t texelOrBlockLength, bool createWithBufferRowLengthPitch = false)
+		{
+			using CONVERSION_FILTER = CConvertFormatImageFilter<inputFormat, outputFormat>;
+			CONVERSION_FILTER convertFilter;
+			CONVERSION_FILTER::state_type state;
+
+			auto createImage = [&](E_FORMAT format, bool copyInputMemory = true)
+			{
+				const auto texelOrBlockByteSize = asset::getTexelOrBlockBytesize(format);
+				const uint32_t pitchTexelOrBlockLength = createWithBufferRowLengthPitch ? calcPitchInBlocks(texelOrBlockLength, texelOrBlockByteSize) : texelOrBlockLength;
+
+				ICPUImage::SCreationParams imgInfo;
+				imgInfo.format = format;
+				imgInfo.type = ICPUImage::ET_1D;
+				imgInfo.extent = { texelOrBlockLength * asset::getBlockDimensions(format).X, 1, 1 };
```

Comment:
wait, what and WHY?

Reply:
it's a single row image

Reply:
it will give it X's texel amount

Reply:
em it should be there https://github.com/buildaworldnet/IrrlichtBAW/pull/468#discussion_r410393907

Reply:
for not BC format's it returns 1 rather, so it;s good I think

Reply:
yeah it returns 1 for not BC, it's good

Reply:
wait why on earth are you making a single row image?

Reply:
cause tga did it before, there is likely a row stream provided by tga and it used convertColor. I had to remake it at least

Reply:
just create an image in the native TGA format, then `convert` at the end, dont do a weird 1D image per row

Reply:
this whole function's existence probably doesn't make sense at all

Reply:
but I need a convert, so filters usage -> I needed create an image

Reply:
TBD... TGA needs its own special review.

this function is used only by TGA right?

Reply:
the function is moved to TGA Loader, since it's specific to it

Reply:
ok moved to TGA